### PR TITLE
Add GenerateKeywordIdeas example

### PR DESCRIPTION
**Pull request by fiboknacky**

### CODE REVIEW

File: examples/Planning/GenerateKeywordIdeas.php [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+        // Generate keyword ideas based on the specified parameters.
+        $response = $keywordPlanIdeaServiceClient->generateKeywordIdeas(
+            $customerId,
+            // Set the language resource using the provided language ID.
+            new StringValue(['value' => ResourceNames::forLanguageConstant($languageId)]),
+            // Add the resource name of each location ID to the request.
+            $geoTargetConstants,
+            // Set the network. To restrict to only Google Search, change the parameter below to
+            // KeywordPlanNetwork.GOOGLE_SEARCH.
```
Comment (jimper) [inline examples/Planning/GenerateKeywordIdeas.php]: KeywordPlanNetwork::GOOGLE_SEARCH?

File: examples/Planning/GenerateKeywordIdeas.php [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            KeywordPlanNetwork::GOOGLE_SEARCH_AND_PARTNERS,
+            $requestOptionalArgs
+        );
+
+        // Iterates over the results and print its detail.
```
Comment (jimper) [inline examples/Planning/GenerateKeywordIdeas.php]: Iterate (singular)

Comment (fiboknacky) [reply examples/Planning/GenerateKeywordIdeas.php]: Done.

File: examples/Planning/GenerateKeywordIdeas.php (hunk 1 above)
Comment (fiboknacky) [reply examples/Planning/GenerateKeywordIdeas.php]: Done.

**Review by jimper — APPROVED** [review]
LGTM

File: examples/Planning/GenerateKeywordIdeas.php [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            // KeywordPlanNetwork::GOOGLE_SEARCH.
+            KeywordPlanNetwork::GOOGLE_SEARCH_AND_PARTNERS,
+            $requestOptionalArgs
+        );
+
+        // Iterate over the results and print its detail.
+        foreach ($response->getResults() as $result) {
+            /** @var GenerateKeywordIdeaResult $result */
+            // Note that the competition printed below is enum value.
+            // For example, a value of 2 will be returned when the competition is 'LOW'.
```
Comment (AnashOommen) [inline examples/Planning/GenerateKeywordIdeas.php]: Thanet, could you pls file a bug against the code generator component to make the rendering of this value better?

File: examples/Planning/GenerateKeywordIdeas.php [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+use Google\Ads\GoogleAds\V0\Enums\KeywordPlanNetworkEnum\KeywordPlanNetwork;
+use Google\Ads\GoogleAds\V0\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V0\Services\GenerateKeywordIdeaResult;
+use Google\Ads\GoogleAds\V0\Services\KeywordAndUrlSeed;
+use Google\Ads\GoogleAds\V0\Services\KeywordSeed;
+use Google\Ads\GoogleAds\V0\Services\UrlSeed;
+use Google\ApiCore\ApiException;
+use Google\Protobuf\StringValue;
+
+/** This example generates keyword ideas from a list of seed keywords. */
```
Comment (AnashOommen) [inline examples/Planning/GenerateKeywordIdeas.php]: or a seed page URL.

File: examples/Planning/GenerateKeywordIdeas.php (hunk 3 above)
Comment (fiboknacky) [reply examples/Planning/GenerateKeywordIdeas.php]: This has been in my radar. 
The proto lib will release a util that helps this soon already. It should be launched soon next year.

**Review by jradcliff — CHANGES_REQUESTED** [review]

File: examples/Planning/GenerateKeywordIdeas.php [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+        // Make sure that keywords and/or page URL were specified. The request must have exactly one
+        // of urlSeed, keywordSeed, or keywordAndUrlSeed set.
+        if (empty($keywords) && is_null($pageUrl)) {
+            throw new \InvalidArgumentException(
+                'At least one of keywords or page URL is required, but neither was specified.'
+            );
+        }
+
+        // Specify the optional arguments of the request as a keywordSeed, UrlSeed,
```
[the remark this answers is not in the document]
Comment (fiboknacky) [reply examples/Planning/GenerateKeywordIdeas.php]: I rather change the other places in this line to lower case to be consistent with the similar sentences in the other line and with the code.

File: examples/Planning/GenerateKeywordIdeas.php [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        $languageId,
+        array $keywords,
+        $pageUrl
+    ) {
+        $keywordPlanIdeaServiceClient = $googleAdsClient->getKeywordPlanIdeaServiceClient();
+
+        // Make sure that keywords and/or page URL were specified. The request must have exactly one
+        // of urlSeed, keywordSeed, or keywordAndUrlSeed set.
+        if (empty($keywords) && is_null($pageUrl)) {
+            throw new \InvalidArgumentException(
```
[the remark this answers is not in the document]
Comment (fiboknacky) [reply examples/Planning/GenerateKeywordIdeas.php]: Yes.

**Review by jradcliff — APPROVED** [review]

**Pull request merged**